src/main/kotlin/exercise5/BoardImpl.kt

```diff
@@ -1,13 +1,74 @@
 package exercise5
 
-// TODO Instantiate SquareBoard
-fun createSquareBoard(width: Int): SquareBoard = TODO("SquareBoardImpl(width)")
+fun createSquareBoard(width: Int): SquareBoard = SquareBoardImpl(width)
 
-// TODO Instantiate GameBoard
-fun <T> createGameBoard(width: Int): GameBoard = TODO("GameBoardImpl(width)")
+fun <T> createGameBoard(width: Int): GameBoard = GameBoardImpl(width)
 
-// TODO Implement SquareBoard Interface
-class SquareBoardImpl(size: Int)
+open class SquareBoardImpl(size: Int) : SquareBoard {
+    override val width: Int = size
+    private val cells: List<Cell>
```

Comment:
This works but it's not performant on the larger `size`'es. Why not use a matrix instead?